lib/base_service.js

```diff
@@ -145,13 +163,16 @@ BaseService.prototype._getOptions = function(opts) {
     args.build = args._.indexOf('build') !== -1 || args.deployRepo;
     args.dockerStart = args._.indexOf('docker-start') !== -1;
     args.dockerTest = args._.indexOf('docker-test') !== -1;
-    args.deployRepo = args.deployRepo || args.build;
+    args.generate = args._.indexOf('generate') !== -1;
+    args.deployRepo = args.deployRepo || args.build || args.building;
 
-    if (args.build && args.dockerStart || args.build && args.dockerTest
-    || args.dockerStart && args.dockerTest) {
+    /*jshint bitwise: false*/
+    if (!(args.build ^ args.dockerStart ^ args.dockerTest ^ args.generate) &&
```

Comment:
Sure, it is a nit-pick & doesn't change functionality. Readability is important, though, and imho the filter version is both shorter and more obvious. It also avoids setting off jshint.